src/middleware_test.ts

```diff
@@ -1,33 +1,36 @@
-import { RequestContext, ConfigurationOptions, HttpMethod } from './gen/index.js';
+import { RequestContext, ConfigurationOptions, HttpMethod, ObservableMiddleware } from './gen/index.js';
 import { deepStrictEqual } from 'node:assert';
 import { setHeaderMiddleware, setHeaderOptions } from './middleware.js';
-import { PromiseMiddleware } from './gen/middleware.js';
 
 describe('Middleware', async () => {
     describe('setHeaderMiddleware', async () => {
         it('should set a header when it is blank', async () => {
             const reqContext = new RequestContext('http://nowhere.com', HttpMethod.GET);
             deepStrictEqual(reqContext.getHeaders(), {});
             const headerMiddleware = setHeaderMiddleware('test-key', 'test-value');
-            const postMiddlewareRequest = await headerMiddleware[0].pre(reqContext);
-            deepStrictEqual(postMiddlewareRequest.getHeaders(), { 'test-key': 'test-value' });
+            const postMiddlewareRequest = await headerMiddleware.pre(reqContext);
+            await postMiddlewareRequest.toPromise().then((request) => {
```

Comment:
Can we avoid mixing `await` and `then()` like this in this file. From a quick glance, it's not immediately obvious what happens if these assertions fail. If it causes an unhandledRejection, I'm not sure if Mocha will report it properly.

Reply:
yes thank you for catching that

Reply:
There seems to be a couple more instances of this pattern. Other than that, this LGTM.